plugins/woocommerce/includes/admin/class-wc-admin-menus.php

```diff
@@ -284,22 +329,41 @@ public function menu_order( $menu_order ) {
 		// Initialize our custom order array.
 		$woocommerce_menu_order = array();
 
+		// Cache HPOS status check.
+		$is_hpos_enabled = wc_get_container()->get( CustomOrdersTableController::class )->custom_orders_table_usage_is_enabled();
```

Comment:
Assuming this HPOS check is needed (related to another question I had about why this conditional check is happening in various places), the cache could be done at the class level? But I also haven't looked at `custom_orders_table_usage_is_enabeld()` to understand whether it's expensive.